src/charm.py

```diff
@@ -196,8 +196,14 @@ def _prometheus_configurations(self):
             configured_jobs.append(job)
 
         alerts = list(self._metrics_providers.alerts().values())
-
-        return {"scrape_jobs": configured_jobs, "alert_rules": alerts[0] if alerts else []}
+        alert_groups = {"groups": []}  # type: ignore
+        for entry in alerts:
+            alert_groups["groups"] = alert_groups["groups"] + entry["groups"]
+
+        return {
+            "scrape_jobs": configured_jobs,
+            "alert_rules": alert_groups if alert_groups["groups"] else {},
+        }
```

Comment:
I think some unit tests around this are crucial.

Reply:
:+1: 

There are no unit tests for any alerting rules at all at the moment.

Reply:
Unit test added. Let me know if you would like anything else tested.